lxd/containers_post.go

```diff
@@ -187,79 +192,63 @@ func createFromMigration(d *Daemon, req *api.ContainersPost) Response {
 		Profiles:     req.Profiles,
 	}
 
-	var c container
-	_, _, err = dbImageGet(d.db, req.Source.BaseImage, false, true)
+	// Grab the container's root device if one is specified
+	storagePool := ""
+	storagePoolProfile := ""
 
-	/* Only create a container from an image if we're going to
-	 * rsync over the top of it. In the case of a better file
-	 * transfer mechanism, let's just use that.
-	 *
-	 * TODO: we could invent some negotiation here, where if the
-	 * source and sink both have the same image, we can clone from
-	 * it, but we have to know before sending the snapshot that
-	 * we're sending the whole thing or just a delta from the
-	 * image, so one extra negotiation round trip is needed. An
-	 * alternative is to move actual container object to a later
-	 * point and just negotiate it over the migration control
-	 * socket. Anyway, it'll happen later :)
-	 */
+	localRootDiskDeviceKey, localRootDiskDevice, _ := containerGetRootDiskDevice(req.Devices)
+	if localRootDiskDeviceKey != "" {
+		storagePool = localRootDiskDevice["pool"]
+	}
 
-	localRootDiskDeviceKey, v := containerGetRootDiskDevice(req.Devices)
-	poolForCopyOrMove := ""
 	// Handle copying/moving between two storage-api LXD instances.
-	// FIXME(brauner): Atm, we do not let users target a specific pool to
-	// move to. So when we receive an invalid/non-existing storage pool, we
-	// simply set it to "" and perform the same pool-searching algorithm as
-	// in the non-storage-api to storage-api LXD instance case seen below.
-	// the container will receive on the remote.
-	if localRootDiskDeviceKey != "" && v["pool"] != "" {
-		_, err := dbStoragePoolGetID(d.db, v["pool"])
+	if storagePool != "" {
+		_, err := dbStoragePoolGetID(d.db, storagePool)
 		if err == NoSuchObjectError {
-			v["pool"] = ""
+			storagePool = ""
 		}
 	}
 
-	// Handle copying or moving containers between a non-storage-api and a
-	// storage-api LXD instance.
-	if localRootDiskDeviceKey == "" || localRootDiskDeviceKey != "" && v["pool"] == "" {
-		// Request came without a root disk device or without the pool
-		// property set. Try to retrieve this information from a
-		// profile. (No need to check for conflicting storage pool
-		// properties in the profiles. This will be handled by
-		// containerCreateInternal().)
+	// If we don't have a valid pool yet, look through profiles
+	if storagePool == "" {
 		for _, pName := range req.Profiles {
 			_, p, err := dbProfileGet(d.db, pName)
 			if err != nil {
 				return InternalError(err)
 			}
 
-			k, v := containerGetRootDiskDevice(p.Devices)
+			k, v, _ := containerGetRootDiskDevice(p.Devices)
 			if k != "" && v["pool"] != "" {
-				poolForCopyOrMove = v["pool"]
-				break
+				// Keep going as we want the last one in the profile chain
```

Comment:
Is there anything in our documentation about how profile devices and container configuration devices are expanded? If not, we should probably add a section somewhere at some point.